### ISSUE TITLE
Add filter to be able to customize the name that displays on the certificate

### ISSUE THREAD

**Issue by donnapep** [woocommerce/sensei-certificates#252]
### Is your feature request related to a problem? Please describe
It's currently not possible to customize the display of the learner's name that appears on the certificate. It displays as _First Name Last Name_ if both are specified, or otherwise uses the learner's display name.

### Describe the solution you'd like
Add a filter so that the learner name can be customized on both the certificate template and the certificate.

**Comment by yscik** [comment]
Could be worth merging #94 for this.

**Comment by cena** [comment]
Not sure if completely related, but user wants to

> add [users'] full name when they finish the course, as our courses are free without any login or any profile

https://wordpress.org/support/topic/users-full-name-can-add-by-them-self-in-no-login-course/#post-14828344

**Comment by m1r0** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add filter to be able to customize the name that displays on the certificate](https://trello.com/c/S01He8zI/1230-add-filter-to-be-able-to-customize-the-name-that-displays-on-the-certificate)

**Comment by m1r0** [comment]
I agree with @yscik. We should consider solving this in a more general way and not only for one data type. Also, this approach most likely will solve these issues: #285, #284, #134, #116, #103, #102, #93.

Thoughts?

cc @yscik @donnapep

**Comment by yscik** [comment]
Yeah let's do it! I looked at it briefly before and there were quite a few merge conflicts IIRC — maybe rebasing that branch onto master is easier since it'd be just one pass.
Also basically the whole thing is duplicated and exists at two places in the code, could be a good opportunity to clean it up :)

**Comment by donnapep** [comment]
@m1r0 and I talked and he's planning to review #94, resolve conflicts and make any changes, if necessary.

**Comment by StefMattana** [comment]
Somehow related to this, mentioned in 5417376-zen: 

>Although I have asked learners to sign in with the name they want on their certificate, they are all using their user name which is not their proper name. If there was a separate way to sign in as a learner only, that would help both the certificate and website user issues.
